light/client.go

```diff
@@ -362,6 +362,13 @@ func (c *Client) initializeWithTrustOptions(ctx context.Context, options TrustOp
 		return fmt.Errorf("invalid commit: %w", err)
 	}
 
+	// Cross-verify with witnesses to ensure everybody have the same view
+	if len(c.witnesses) > 0 {
+		if err := c.compareFirstHeaderWithWitnesses(ctx, l.SignedHeader); err != nil {
```

Comment:
I think we should actually do this check on line 227. If we initialize from a trusted store we should still check the witnesses. They might have even changed

Reply:
agree. but this will require adding context to `NewClientFromTrustedStore`

Reply:
^ Is this much of a problem?

Reply:
I'm a bit hesitant to introduce network calls into NewClientFromTrustedStore. I think it's fine if we only check very first header. Because only in that case we can't generate any evidence, right? In all other cases (when we have a trusted header), any verification attempt will lead to light client creating a trace and submitting it to both primary and witness.

Reply:
yeah okay I think it will be fine if you leave it as is. If it's from a trusted store then unless we've changed the witnesses in between (which could be the case) then they should have already been verified previously